poller/poller.go

```diff
@@ -105,10 +106,12 @@ func (p *Poller) Poll(ctx context.Context, n int, id string, interval time.Durat
 				return
 			case <-pollTimer.C:
 				logrus.WithField("delegate_id", id).Info("polling for tasks")
-				tasks, err := p.Client.GetTaskEvents(ctx, id)
+				taskEventsCtx, cancelFn := context.WithTimeout(ctx, taskEventsTimeout)
+				tasks, err := p.Client.GetTaskEvents(taskEventsCtx, id)
 				if err != nil {
 					logrus.WithError(err).Errorf("could not query for task events")
 				}
+				cancelFn()
```

Comment:
Where is this cancelFn defined?

Reply:
L109 return. From documentation, once a context is not being used - we should cancel it so that it frees up resources

Reply:
actually, this can be moved out of the loop - i will update that